### ISSUE TITLE
Serialize frozen objects correctly

### ISSUE THREAD

**Issue by NTillmann** [facebook/prepack#953]
Currently, for
```js
(function () {
  var o = { x: 23, y: 42 };
  Object.freeze(o);
  global.inspect = function() { o.z = "bug"; return o.z; }
})();
```

we generate the following code.
```js
(function () {
  var $$0 = {
    enumerable: true,
    configurable: false,
    writable: false
  };

  var _$0 = this;

  var _$1 = _$0.Object;
  var _$2 = _$1.defineProperty;

  var _0 = function () {
    _1.z = "bug";
    return _1.z;
  };

  var _1 = {};
  $$0.value = 23;

  _$2(_1, "x", $$0);

  $$0.value = 42;

  _$2(_1, "y", $$0);

  inspect = _0;
}).call(this);
```

The serialized code is...
- incorrect, as the serialized Object never gets frozen. If you drop in the code at the top as a new serializer test case, you'll see the failure, and
- terribly verbose.

Ideally, for this simple example, we'd generate code that's very similar to the original code.

To this end, if an object is frozen (integrity level "frozen"), then the serializer should...
- emit all object properties as regular properties, ignoring the flags (no calls to `Object.defineProperty`)
- then emit a call to `Object.freeze`.

**Comment by NTillmann** [comment]
This is related to #585.

**Comment by echo304** [comment]
Let me tackle this one ✋

**Comment by NTillmann** [comment]
Hi @echo304, do you think you'll have time to work on this, or can I look for other volunteers? Thanks!

**Comment by echo304** [comment]
Sorry @NTillmann ! You'd better find other awesome guy who is able to solve this issue!

**Comment by NTillmann** [comment]
Addressed by #1436.
